app_dart/lib/src/request_handlers/push_gold_status_to_github.dart

```diff
@@ -0,0 +1,307 @@
+            log.debug('Notifying for triage.');
+            await _commentAndApplyGoldLabel(gitHubClient, pr, slug);
+          }
+        }
+      } else {
+        // We have seen this commit before.
+        // If checks are still running, or last update was green, do nothing.
+        if (cirrusCheckStatuses.any(cirrusInProgressStates.contains)
+          || lastUpdate.status == GithubGoldStatusUpdate.statusCompleted)
+          return Body.empty;
+
+        // Check Gold for new status and update.
+        final String status = await _getGoldStatus(pr, log);
+        log.debug('Checks are completed, Gold reports $status status for '
+          '${pr.number} sha ${pr.head.sha}.');
+        if (lastUpdate.status != status) {
+          // The 'running' state is used when the check is waiting and when
+          // triage is needed, so if it is already 'running' we don't need to
+          // update it. The comment feature below alerts the author that the
+          // result needs to be addressed.
+          statusRequest = CreateStatus(status);
+          statusRequest.targetUrl = kTargetUrl;
+          statusRequest.context = kContext;
+          statusRequest.description = _getStatusDescription(status);
+        }
+        if (status == GithubGoldStatusUpdate.statusRunning
+          && !await _alreadyCommented(gitHubClient, pr, slug)) {
+          log.debug('Notifying for triage.');
+          await _commentAndApplyGoldLabel(gitHubClient, pr, slug);
+        }
+      }
+
+      if (statusRequest != null) {
+        try {
+          await gitHubClient.repositories.createStatus(slug, pr.head.sha, statusRequest);
+          lastUpdate.status = statusRequest.state;
+          lastUpdate.head = pr.head.sha;
+          lastUpdate.updates += 1;
+          statusUpdates.add(lastUpdate);
+        } catch (error) {
+          log.error(
+            'Failed to post status update to ${slug.fullName}#${pr.number}: $error');
+        }
+      }
+    }
+
+    final int maxEntityGroups = config.maxEntityGroups;
+    for (int i = 0; i < statusUpdates.length; i += maxEntityGroups) {
+      await datastore.db.withTransaction<void>((Transaction transaction) async {
+        transaction.queueMutations(
+          inserts: statusUpdates.skip(i).take(maxEntityGroups).toList());
+        await transaction.commit();
+      });
+    }
+    log.debug('Committed all updates');
+    return Body.empty;
+  }
+
+  /// Queries the Cirrus GraphQL for build information for the given pr and sha.
+  Future<List<dynamic>> _queryGraphQL(
+    String pr,
+    String sha,
+    GraphQLClient client,
+  ) async {
+    const String owner = 'flutter';
+    const String name = 'flutter';
+    const String cirrusStatusQuery = r'''
+    query BuildBySHAQuery($owner: String!, $name: String!, $SHA: String) { 
+      searchBuilds(repositoryOwner: $owner, repositoryName: $name, SHA: $SHA) {
+        pullRequest(first: 1, query: $PR) { 
```

Comment:
@fkorotkov can you confirm this? Thanks!

Reply:
LGTM if you tested to query at least once 😅

Reply:
Is there an explorer like https://developer.github.com/v4/explorer/ where I can test the query?

Reply:
Will add one to Cirrus shortly https://github.com/cirruslabs/cirrus-ci-web/pull/154

In the meantime you can `brew install graphiql` locally and point to https://api.cirrus-ci.com/graphql